### PR TITLE
Add oauth authorize endpoint implementation with error handling

### PR DESCRIPTION
**Pull request by dannyhayes-gds**
Fix the public/assets paths and refactor the validation handling.

Co-authored-by: Tal Nagra <tal.nagra@digital.cabinet-office.gov.uk>

### What changed
Created the /authorize oauth endpoint for the credential issuer.

### Why did it change
Allows the ipv core to initiate an oath flow and request an authorisation code.

### Issue tracking

- [PYI-357](https://govukverify.atlassian.net/browse/PYI-357)

### Environment variables or secrets

- [x] No environment variables or secrets were added or changed

### Other considerations

- [x] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,3 +12,8 @@ The Orchestrator Stub allows the user to select desired attributes then initiate
 
 On completion of the user journey the Orchestrator Stub receives an authorisation code which it will exchange for an access token and in turn use to access the protected resource in the IPV system. Finally the Orchestrator Stub will display the contents of the protected resource.
 
+## Credential Issuer Stub
+`di-ipv-credential-issuer-stub` [/di-ipv-credential-issuer-stub](/di-ipv-credential-issuer-stub)
+
+The Credential issuer stub allows the user to start an Oauth flow in order to access the credential issuers protected resource.
```
Comment (kerrr) [inline README.md]: This isn't really the stub's primary mission. 
```suggestion
The Credential Issuer Stub can be used as an endpoint for testing Credential Issuer OAuth flows from the core IPV system. It provides an '/authorize' user endpoint, token exchange and access to a protected resource.
```

File: di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/handlers/QueryStringParamConstants.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package uk.gov.di.ipv.stub.cred.handlers;
+
+public class QueryStringParamConstants {
```
Comment (kerrr) [inline di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/handlers/QueryStringParamConstants.java:3]: Does it add value to have this as a separate class? Do these values just belong in `AuthorizeHandler`?

File: di-ipv-credential-issuer-stub/src/main/resources/ErrorMessages.properties [hunk 3]
```diff
@@ -0,0 +1,4 @@
+missing_redirect_uri=redirect_uri param must be provided
+missing_response_type=response_type param must be provided
+invalid_response_type=response_type param must be set to 'code'
+missing_client_id=client_id param must be provided
```
Comment (kerrr) [inline di-ipv-credential-issuer-stub/src/main/resources/ErrorMessages.properties]: Please have EOF on it's own line.

File: di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/handlers/QueryStringParamConstants.java (hunk 2 above)
Comment (dannyhayes-gds) [reply di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/handlers/QueryStringParamConstants.java:3]: We moved them here into its own class as they are going to be used within the other endpoint classes

File: README.md (hunk 1 above)
Comment (dannyhayes-gds) [reply README.md]: Sounds good, I'll make this change this morning.

**Review by kerrr — APPROVED** [review]
🚀

**Pull request merged**